### ISSUE TITLE
Issue with SheepShaver MacOSX 64-bit in full-screen mode in OSX 10.6

### ISSUE THREAD

**Issue by RonaldPR** [cebix/macemu#4]
This issue exists only with SheepShaver 64-bit in full-screen mode in Snow Leopard

It does not exist in window mode, not with SheepShaver 32-bit, and not in Lion.

SheepShaver 64-bit in full-screen mode in Snow Leopard is all but unusable. The MacOS menu bar in the emulated machine cannot be used. When you click in the menu bar, the menus do not work and a cursor image gets stuck on the menu bar. The only way out is force-quitting SheepShaver.

**Comment by kencu** [comment]
For me this happens with 64bit builds of both SheepShaver and BasiliskII using the latest (2017-12-15) sources. Switching back to windowed mode makes everything OK again.

**Comment by asvitkine** [comment]
I'm not seeing this issue with the 64-bit SS on Snow Leopard, compiled from
the Xcode project from tip of tree and SDL 1.2.14. Are you using some
different settings?

**Comment by kencu** [comment]
I just rebuilt it again to check, and same thing happens - 64bit only. I'm on Snow Leopard as well. 

Yes, I am building it from the configure / Unix args, using the MacPorts portfile. If you use MacPorts, you can install the default 32bit version of sheepshaver with `sudo port install sheepshaver`. Seems to work perfectly.

To change to the 64bit build takes a few seconds. Uninstall sheepshaver (`sudo port uninstall sheepshaver`), then edit the Portfile with any editor

bbedit `port file sheepshaver`

find the line

`supported_archs   i386`

and change it to 

`supported_archs   x86_64`

If you now `sudo port install sheepshaver` you should get the 64bit version, and it shows the menu bar error described in this issue.

You can also see the other configure settings I'm using there. It might have something to do with compiler choices, etc, tho. I'm presently building it with clang-3.9 myself.

**Comment by asvitkine** [comment]
I am able to reproduce the issue using MacPorts.

Note: I didn't have to change the port file settings - it installed a
64-bit one by itself? (Or maybe I had out-of-date port definitions? How do
I check?)

So it must be some difference in the configuration settings. Definitely
building with clang-3.9 is different than what Xcode project does - since
that one rebuilds the JIT generated code - rather than using the committed
versions of those. I also noticed MacPorts is using the GTK UI, rather than
the Cocoa one. I guess I should try to replicate those build settings
locally and debug it.

**Comment by asvitkine** [comment]
Yeah my port files were out of date. Anyway, I am able to repro the issue
by building from the repo with:

./configure --disable-vosf  --without-esd --without-mon --enable-sdl-video
--enable-sdl-audio --enable-jit-compiler --enable-standalone-gui
--with-gtk=no

So the question is what's different between the two builds.

**Comment by kencu** [comment]
Glad we're on the same page! That makes fixing it much more likely.

You can see if you have the current version by  `port -v installed sheepshaver`
which shows me:

The following ports are currently installed:
  sheepshaver @20171215_0 (active) platform='darwin 10' archs='i386' date='2017-12-22T10:07:35-0800'

You should get the i386 build by default - I set the portfile that way. If you don't, please

sudo port uninstall sheepshaver

sudo port selfupdate

sudo port -v install sheepshaver

and you should get it.

To force a build from source on MacPorts, rather than downloading a prebuilt binary, do

sudo port -v -s install sheepshaver

I built it with the GTK2 standalone GUI, but the cocoa settings editor is also embedded in the sheepshaver app, available under "Preferences".

**Comment by asvitkine** [comment]
I see the issue by using the checked-in x86_64 config.h file that's used by
Xcode project as the config.h and building locally. So it's nothing at the
source code level.

My guess is this must be a difference between the MacPorts SDL vs. the SDL
framework (downloaded from SDL site) that the Xcode build uses.
